awx/main/isolated/manager.py

```diff
@@ -304,7 +303,7 @@ def consume_events(self, dispatcher):
                         # again on the next sync
                         pass
                     event_data.setdefault(self.event_data_key, self.instance.id)
-                    dispatcher.dispatch(event_data)
+                    self.event_handler(event_data)
```

Comment:
I'm missing something. Right below this, it does handle artifacts. But the event_callback method that you're trying to pull in handles artifacts.

https://github.com/ansible/awx/blob/46ad5b6a9ad5ac3e2b66c545a72d33e39e8b50c0/awx/main/tasks.py#L1171-L1177

I think it's fine to lose the version that's here, just a few lines below here.

Unrelated - looking at this makes me a little bit worried about the size of this `self.handled_events`, but that's maybe a problem for another time.

Reply:
Keep in mind that `handled_events` is a list of file names not *actual* events.

```>>> len('/tmp/awx_100_abcdefg/artifacts/100/events/uuid-uuid-uuid-uuid.json') * 500000
33000000```

30MB?  Not _ideal_, but not catastrophic.  And it's going to garbage collected when the reference is destroyed.

Reply:
Good eye, I've uploaded a change to get rid of this extra code.